src/Inventory/Asset/InventoryAsset.php

```diff
@@ -471,7 +471,23 @@ protected function handleInput(\stdClass $value, ?CommonDBTM $item = null): arra
                     $input[$key] = $this->known_links[$known_key];
                     $input['_raw' . $key] = $this->raw_links[$known_key];
                 } else {
-                    $input[$key] = $this->raw_links[$known_key];
+                    // If $item is new and the input key is locked, we do not want to set it using the raw value.
+                    // This is because locked fields are no longer processed or sanitized during the addition process.
+                    // For more details, see: https://github.com/glpi-project/glpi/pull/19426
+                    // Bypass for :
+                    // - states_id if default value is define from inventory conf
+                    // - entities_id default inventory conf is 0 'root entity'
+                    $config = \Config::getConfigurationValues('inventory');
+
+                    if (
+                        !$item->isNewItem()
+                        || (
+                            ($key === 'states_id' && !empty($config['states_id_default']))
```

Comment:
Why not populating `$this->known_links` with the known key extracted from the `states_id_default` config ?